testbed/tests/metric_test.go

```diff
@@ -79,3 +91,215 @@ func TestMetric10kDPS(t *testing.T) {
 	}
 
 }
+
+const (
+	mockedConsumedResourceWithType          = "mocked_consumed_resource_with_type"
+	mockedConsumedResourceWithoutType       = "mocked_consumed_resource_without_type"
+	mockedConsumedResourceNil               = "mocked_consumed_resource_nil"
+	mockedConsumedResourceWithoutAttributes = "mocked_consumed_resource_without_attributes"
+)
+
+func TestMetricResourceProcessor(t *testing.T) {
+	inputs, err := ioutil.ReadFile("testdata/resource-processor-inputs.json")
+	require.NoError(t, err, "failed to load test dependencies", err)
+
+	mockedConsumedResources := map[string]*otlpmetrics.ResourceMetrics{}
+	err = json.Unmarshal(inputs, &mockedConsumedResources)
+	require.NoError(t, err, "failed to load test dependencies", err)
+	require.Equal(t, 4, len(mockedConsumedResources), "not all mocked resources loaded")
+
+	sender := testbed.NewOTLPMetricDataSender(testbed.GetAvailablePort(t))
+	receiver := testbed.NewOTLPDataReceiver(testbed.GetAvailablePort(t))
+
+	tests := []struct {
```

Comment:
The only part that I am a bit wary about is that these 4 test cases correspond to 4 json definitions but they are disconnected and are in different files. Can we do this slightly differently to ensure each test case and its corresponding json are near each other? For example have a separate *.go file which contains this data, including the json as a Go string and eliminate the json file.

Reply:
Updated to have resource metric definitions as string constants along with the test cases, eliminating the JSON file. Let me know if this seems to be a better approach.